src/conf.c

```diff
@@ -135,8 +135,10 @@ static int validate_common(cfg_t *cfg, const char *provider, int custom)
 	}
 
 	if (!custom && !cfg_getstr(cfg, "username")) {
-		cfg_error(cfg, "Missing username setting for DDNS provider %s", provider);
-		return -1;
+		if (strcmp(provider, "default@freemyip.com") != 0) {	
```

Comment:
Btw, would you care to add your actual name to the Copyright of the plugin?  Didn't notice it at first, and it's just a formality ...